File: ruby/j11.arguments.rb
#!/usr/bin/ruby
arg_count = ARGV.length # get num of arguments
first     = 0           # set index of first element
count     = 0           # initialize counter

puts "The arugments passed are:"
while ARGV.length >= 1 do
 puts " item #{count+=1}: #{ARGV[first]}" # output result of first argument
 ARGV.shift                               # remove first element
end

File: php/j11.arguments.php
#!/usr/bin/php
<?php
$count       = 1;        // set initial counter
$script_name = $argv[0]; // get script name

echo "The arguments passed are:\n";
// collection loop to enumerate args
foreach ($argv as $arg) {
  // skip if we get name of the script 
  if ("$arg" == "$script_name" ) continue;
  // output count and argument
  echo "  item $count: $arg\n";
  $count++;              // increment counter
}
?>